### PR TITLE
fix(card-and-divider-padding): added padding to divider + section

### LINKED ISSUE #2698: Update: `pf-c-card__body` padding
Screen Shot 2020-02-11 at 12 54 32 PM

`.pf-c-divider` in `.pf-c-card` doesn't work as expected. If we update `--pf-c-card__body--PaddingTop` to `0` only when it's a sibling of another `card__head`, `card__header`, `card__body` or `card__footer`, then `.pf-c-divider` will work as expected when added.

Would look something like this;
```
.pf-c-card__head,
.pf-c-card__header,
.pf-c-card__body,
.pf-c-card__footer {
  padding-top: var(--pf-c-card--first-child--PaddingTop);
  padding-right: var(--pf-c-card--child--PaddingRight);
  padding-bottom: var(--pf-c-card--child--PaddingBottom);
  padding-left: var(--pf-c-card--child--PaddingLeft);

  & + & {
    --pf-c-card--first-child--PaddingTop: 0;
  }
}
```

` --pf-c-card--first-child--PaddingTop` would be updated to `--pf-c-card--PaddingTop` during breakpoint change.

### PR DESCRIPTION
**Pull request by mattnolting**
closes #2698

### CODE REVIEW

**Comment by mcoker** [comment]
@mceledonia @mcarrano when you get a sec, would you mind looking at this? It adds a divider between card sections, and inserts the divider after the default padding between sections, then adds more padding. So it adds a divider + 16px spacing after a section.

I'm curious if that's how we want the spacing to be when there is a divider between sections, or should the divider go in between the existing space between sections with no addition to the existing spacing?

**Comment by mcarrano** [comment]
Looks good to me @mcoker . But I'll defer to @mceledonia on your questions above.

**Comment by mcoker** [comment]
ping @mceledonia

**Comment by mattnolting** [comment]
@mceledonia ping, wdyt??

**Comment by mceledonia** [comment]
This looks good to me, heads up any `compact` variants (e.g. card) should probably drop down to --md spacer.

**Review by mceledonia — APPROVED** [review]

File: src/patternfly/components/Card/card.scss [hunk 1]
```diff
@@ -99,6 +102,15 @@
       transform: rotate(var(--pf-c-card--m-expanded__header-toggle-icon--Rotate));
     }
   }
+
+  > .pf-c-divider {
+    & + .pf-c-card__header,
```
Comment (mcoker) [inline src/patternfly/components/Card/card.scss:107]: I don't think you need the `&`?

**Review by mcoker — APPROVED** [review]
LGTM! One nit about the `&` but it's fine as is.

**Pull request merged**